### DISCUSSION TITLE
How to do `calc` with utilities?

### DISCUSSION THREAD

**Discussion by JeromeDeLeon** [windicss/windicss#340] [Q&A] [answered]
How do you achieve the calculation with the class name or is that even possible?

```html
<nav class="h-16">
</nav>
<div style="height: calc(h-screen - h-16);">
</div>
```

**Comment by shiftgeist** [comment, marked as answer]
I think `calc` is not intended to be used in templates (HTML).

Another way that should work:

```diff
<nav class="h-16">
</nav>
- <div style="height: calc(h-screen - h-16);">
+ <div class="h-screen p-16"> 
</div>
```

I you want to add a custom class you could do so:

```js
// windi.config.js
export default {
  theme: {
    extend: {
      height: (theme) => ({
        'h-screen-minus-16': `calc(${theme('height.screen')} - ${theme('height.16')}`
      }),
    },
  },
}
```

Please note that I wrote these off the top of my head and didn't test them.

**Reply by JeromeDeLeon** [reply]
The `style`  I put there was only meant to be for showcasing what I want to achieve with `windicss` cuz I thought it would be possible like `h-[calc(100vh_-_util('h-16'))]` or `h-[calc(util('h-screen')_-_util('h-16'))]` but I guess I was too futuristic 😅
